cpp/base/notification.cc

```diff
@@ -9,16 +9,20 @@ using std::unique_lock;
 
 namespace cert_trans {
 
+Notification::~Notification() {
+  // Ensure nobody is inside Notify() when we're destructing.
```

Comment:
Remove that bit, as we know it's not helping? I'd rather not artificially hide even more things...